### PR TITLE
feat(job-runner): record job's start timestamp to support timeouts

### PR DESCRIPTION
**Pull request by xurui-c**
store a record of when a job started executing so that we can support timeouts

### CODE REVIEW

File: snuba/manual_jobs/runner.py [hunk 1]
```diff
@@ -128,9 +129,11 @@ def run_job(job_spec: JobSpec) -> JobStatus:
...
                 job_spec.job_id, JobStatus.ASYNC_RUNNING_BACKGROUND
             )
+            _record_start_time(job_spec.job_id)
             job_to_run.execute(job_logger)
         else:
             current_job_status = _set_job_status(job_spec.job_id, JobStatus.RUNNING)
+            _record_start_time(job_spec.job_id)
             job_to_run.execute(job_logger)
             current_job_status = _set_job_status(job_spec.job_id, JobStatus.FINISHED)
             job_logger.info("[runner] job execution finished")
```
Comment (onkar) [inline snuba/manual_jobs/runner.py:139]: We are doing pretty much the same things in the `if` and `else` blocks here. The difference is just the args passed. Can this be simplified?:
```
job_running = JobStatus.ASYNC_RUNNING_BACKGROUND if job_spec.is_async else JobStatus.RUNNING
try:
    current_job_status = _set_job_status(job_spec.job_id, job_running)
    _record_start_time(job_spec.job_id)
    job_to_run.execute(job_logger)
    if not job_spec.is_async:
        # Finish current job if no exception is raised and it is synchronous.
        current_job_status = _set_job_status(job_spec.job_id, JobStatus.FINISHED)
        job_logger.info("[runner] job execution finished")
except BaseException:
    ...
```

**Review by onkar — APPROVED** [review]
LGTM

File: snuba/manual_jobs/runner.py (hunk 1 above)
Comment (xurui-c) [reply snuba/manual_jobs/runner.py]: This approach is much cleaner. Thank you Onkar!

**Review by kylemumma — APPROVED** [review]
nice

File: tests/manual_jobs/test_record_job_start.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+from unittest.mock import patch
```
Comment (onewland) [inline tests/manual_jobs/test_record_job_start.py:1]: In the future, I think a method this narrow doesn't need its own test file, but it's fine for now

**Review by onewland — APPROVED** [review]

**Pull request merged**